src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -250,16 +251,40 @@ export class WorkspaceSavedObjectsClientWrapper {
       objects: Array<SavedObjectsBulkCreateObject<T>>,
       options: SavedObjectsCreateOptions = {}
     ): Promise<SavedObjectsBulkResponse<T>> => {
-      if (options?.workspaces && options.workspaces.length > 0) {
-        const permitted = await this.validateMultiWorkspacesPermissions(
-          options.workspaces,
+      const isPassedWorkspaces = options?.workspaces && options.workspaces.length > 0;
+
+      if (
+        isPassedWorkspaces &&
+        !(await this.validateMultiWorkspacesPermissions(
+          options.workspaces!,
           wrapperOptions.request,
           [WorkspacePermissionMode.LibraryWrite, WorkspacePermissionMode.Management]
-        );
-        if (!permitted) {
-          throw generateSavedObjectsPermissionError();
+        ))
+      ) {
+        throw generateSavedObjectsPermissionError();
+      }
+
+      if (options.overwrite) {
+        for (const object of objects) {
+          const { type, id } = object;
+          if (
+            id &&
+            !(await this.validateWorkspacesAndSavedObjectsPermissions(
+              await wrapperOptions.client.get(type, id),
```

Comment:
@SuZhou-Joe What do you think  about if we need to check passed `object.id` was exists?

Reply:
@SuZhou-Joe  Could you take a look? Do you have any comments?

Reply:
If no id is provided, the bulkCreate will create brand new object inside `options.workspaces`. So I'd recommend that.
1. If id exists, do the permission check shown above.
2. If id not exist, check the library_write permission on the `options.workspaces`.

Reply:
I think we need to check if the object of provided id exists first. If we can't get this object, I think we can ignore the permission validation here. This case is for bulkCreate with override option, but passed some not exists objects. We need to skip permission validation for these objects. Or the bulkCreate will throw an not found error.

Reply:
> This case is for bulkCreate with override option, but passed some not exists objects. We need to skip permission validation for these objects. Or the bulkCreate will throw an not found error.

Would it be more valid if we put this logic in caller side or still in client wrapper ?

Reply:
In my opinion, pass brand new objects here with override option is a valid case here. Throw object not found error will deny this behavior.